<p>Loop counters should not be modified in the body of the loop. However other loop control variables representing logical values may be modified in
the loop, for example a flag to indicate that something has been completed, which is then tested in the for statement.</p>
<h2>Noncompliant Code Example</h2>
<pre>
var names = new string[]{ "Jack", "Jim", null, "John" };
for (var i = 0; i &lt; names.length; i++) {
  if (names[i] == null) {
    i = names.length;                         // Non-Compliant
  } else {
    Console.WriteLine(names[i]);
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
var names = new string[] { "Jack", "Jim", null, "John" };
for (var i = 0; i &lt; names.length; i++) {
  if (names[i] == null) {
    break;
  }
  Console.WriteLine(names[i]);
}
</pre>

